File: backend/src/main/java/com/back/domain/chat/dto/response/ChatMessageResponseDto.java
package com.back.domain.chat.dto.response;

import com.back.domain.chat.entity.ChatMessage;
import java.time.LocalDateTime;
import lombok.Builder;

@Builder
public record ChatMessageResponseDto(
        Long messageId,
        Long roomId,
        Long senderId,
        String senderName,
        String content,
        LocalDateTime sentAt
) {
    public static ChatMessageResponseDto from(ChatMessage chatMessage) {
        return ChatMessageResponseDto.builder()
                .messageId(chatMessage.getId())
                .roomId(chatMessage.getChatRoom().getId())
                .senderId(chatMessage.getSender().getId())
                .senderName(chatMessage.getSender().getName())
                .content(chatMessage.getContent())
                .sentAt(chatMessage.getSentAt())
                .build();
    }
} 

File: backend/src/main/java/com/back/domain/chat/dto/response/ChatMessageResponseDto.kt
package com.back.domain.chat.dto.response

import com.back.domain.chat.entity.ChatMessage
import java.time.LocalDateTime

data class ChatMessageResponseDto(
    val messageId: Long?,
    val roomId: Long?,
    val senderId: Long?,
    val senderName: String?,
    val content: String?,
    val sentAt: LocalDateTime?
) {
    companion object {
        fun from(chatMessage: ChatMessage): ChatMessageResponseDto {
            return ChatMessageResponseDto(
                messageId = chatMessage.id,
                roomId = chatMessage.chatRoom.id,
                senderId = chatMessage.sender.id,
                senderName = chatMessage.sender.name,
                content = chatMessage.content,
                sentAt = chatMessage.sentAt
            )
        }
    }
}